pyo3-macros-backend/src/pyimpl.rs

```diff
@@ -84,6 +85,17 @@ pub fn build_py_methods(
     }
 }
 
+fn check_pyfunction(meth: &ImplItemFn) -> syn::Result<()> {
+    if meth
+        .attrs
+        .iter()
+        .any(|attr| attr.path().is_ident("pyfunction"))
```

Comment:
I think if you make this check remove the attribute rather than just check for it, it will improve the error output by removing all the gibberish related to the pyfunction expansion.

Also, see what we recently did in #4288 - we should check for the the "namespaced" attribute here too.

Reply:
Hi David,

Thanks for the review!

I added clearing `attrs` and it removes the other error messages, but honestly I'm not sure if this is the correct solution. What do you think?

Regarding the namespaces, I'm reading the PR you linked.

Thanks again,
Zsolt

ps: this PR was created during europython 2024 sprint weekend.

Reply:
I was thinking about namespaces and I found the `has_attribute` function, so at the end the "if" can be updated to:
```rust
if has_attribute(&meth.attrs, "pyfunction")`
```

But this function is not public so I cannot use it in pyimpl.rs. I checked the references and it seems no code path runs through `module.rs`.

So if this is the way to solve (I mean, calling `has_attribute`). I think it can be done by:
* making this function public
* moving this function to some common source file (and making it public there).

Or something else. What do you think?

Zsolt

Reply:
I think you want both `has_attribute` and `has_attribute_with_namespace`. Perhaps move them to `utils.rs` and make them `pub(crate)`?

Reply:
I moved both functions, and I also had to move

```rust
pub(crate) enum IdentOrStr<'a> {
    Str(&'a str),
    Ident(syn::Ident),
}
```

as well to `utils.rs`.